ctapipe/reco/sklearn.py

```diff
@@ -507,6 +501,318 @@ def predict_table(self, key, table: Table) -> Table:
+            raise IOError(f"Path {path} exists and overwrite=False")
+
+        with path.open("wb") as f:
+            Provenance().add_output_file(path, role="ml-models")
+            joblib.dump(self, f, compress=True)
+
+    @classmethod
+    def read(cls, path, **kwargs):
+        with open(path, "rb") as f:
+            instance = joblib.load(f)
+
+        for attr, value in kwargs.items():
+            setattr(instance, attr, value)
+
+        if not isinstance(instance, cls):
+            raise TypeError(
+                f"{path} did not contain an instance of {cls}, got {instance}"
+            )
+
+        Provenance().add_input_file(path, role="ml-models")
+        return instance
+
+    @lazyproperty
+    def instrument_table(self):
+        return self.subarray.to_table("joined")
+
+    def _predict(self, key, table):
+        if key not in self._models:
+            raise KeyError(
+                f"No model available for key {key},"
+                f" available models: {self._models.keys()}"
+            )
+        X, valid = table_to_X(table, self.features, self.log)
+        prediction = np.full(len(table), np.nan)
+
+        if np.any(valid):
+            valid_norms = self._models[key][0].predict(X)
+
+            if self.log_target:
+                prediction[valid] = np.exp(valid_norms)
+            else:
+                prediction[valid] = valid_norms
+
+            prediction[valid] *= self._models[key][1].predict(X)
+
+        if self.unit is not None:
+            prediction = u.Quantity(prediction, self.unit, copy=False)
+
+        return prediction, valid
+
+    def __call__(self, event: ArrayEventContainer) -> None:
+        """Event-wise prediction for the EventSource-Loop.
+
+        Fills the event.dl2.tel[tel_id].disp[prefix] container
+        and event.dl2.tel[tel_id].geometry[prefix] container.
+
+        Parameters
+        ----------
+        event: ArrayEventContainer
+        """
+        for tel_id in event.trigger.tels_with_trigger:
+            table = collect_features(event, tel_id, self.instrument_table)
+            table = self.feature_generator(table)
+
+            passes_quality_checks = self.quality_query.get_table_mask(table)[0]
+
+            if passes_quality_checks:
+                disp, valid = self._predict(self.subarray.tel[tel_id], table)
+                disp_container = DispContainer(
+                    norm=disp[0],
+                    is_valid=valid[0],
+                )
+
+                if valid:
+                    hillas = event.dl1.tel[tel_id].parameters.hillas
+                    psi = hillas.psi.to_value(u.rad)
+
+                    fov_lon = hillas.fov_lon + disp * np.cos(psi)
+                    fov_lat = hillas.fov_lat + disp * np.sin(psi)
+                    alt, az = telescope_to_horizontal(
+                        lon=fov_lon,
+                        lat=fov_lat,
+                        pointing_alt=event.pointing.tel[tel_id].altitude.to(u.deg),
```

Comment:
Unnecessary conversion. Either convert `to_value` if something unit-less is required or drop the `.to(u.deg)`.

Reply:
Also, using the table method here instead of the normal transformation seems strange. I'd rather see the `TelescopeFrame(...).transform_to(AltAz())`

Reply:
Done